### ISSUE TITLE
[BUG] discussion channel

### ISSUE THREAD

**Issue by Maurowin** [azerothcore/azerothcore-wotlk#2794]
##### SMALL DESCRIPTION:

When we create a new character, we are not directly included in channels /1 and /3 except for the races :
- Undead
- Tauren

##### EXPECTED BLIZZLIKE BEHAVIOUR:

From the first connection of a newly created character, it must integrate the /1 and /3 discussion channels.

##### CURRENT BEHAVIOUR:

When we create a new character, we are not directly included in channels /1 and /3 except for the races :
- Undead
- Tauren
This problem only occurs the first time the character connects.

##### STEPS TO REPRODUCE THE PROBLEM:

1. create new charecter (no undead and taurent)
2. connect first time (skip or not introduction video)
3. look your chat

##### BRANCH(ES):

master

**Comment by Si1ker** [comment]
**Not confirmed.** / working well.

I got 1 and 3 as normal for all races.

![draeneiyes](https://user-images.githubusercontent.com/61223313/77384585-64d6e080-6d4b-11ea-861e-8fd22b241827.png)
![elfyes](https://user-images.githubusercontent.com/61223313/77384586-656f7700-6d4b-11ea-8f65-c925a1668ffa.png)
![enanoyes](https://user-images.githubusercontent.com/61223313/77384587-656f7700-6d4b-11ea-96b6-e03462dea05f.png)
![gnomeyes](https://user-images.githubusercontent.com/61223313/77384588-66080d80-6d4b-11ea-923c-540ce6aaa2c7.png)
![humanyes](https://user-images.githubusercontent.com/61223313/77384589-66080d80-6d4b-11ea-9e24-2b8fa2d63b9d.png)
![nocelfyes](https://user-images.githubusercontent.com/61223313/77384590-66080d80-6d4b-11ea-9015-38199bacf175.png)
![orcyes](https://user-images.githubusercontent.com/61223313/77384592-66080d80-6d4b-11ea-9968-85f395273918.png)
![taurenyes](https://user-images.githubusercontent.com/61223313/77384594-66a0a400-6d4b-11ea-9d8b-7e9556b4de11.png)
![trolyes](https://user-images.githubusercontent.com/61223313/77384596-66a0a400-6d4b-11ea-840e-53fa6ab55eda.png)
![undeadyes](https://user-images.githubusercontent.com/61223313/77384597-66a0a400-6d4b-11ea-881d-83ebb6f7cf55.png)


I got all channels when in the town/city.
![all](https://user-images.githubusercontent.com/61223313/77384663-8df77100-6d4b-11ea-8142-56a151862efa.png)


AC HASH/COMMIT: 846f786

**Comment by Maurowin** [comment]
Do u see something on the chat box?

**Comment by Maurowin** [comment]
wait before close, i will make a video.

**Comment by solidmaxtor** [comment]
I only confirm for first login not autojoin all channels, if player relogin join normal.
For  @Si1ker images this is a normal mode no bug. Comerce, lfgroup, lfguild channels only work in city.

**Comment by Maurowin** [comment]
> 
> 
> I only confirm for first login not autojoin all channels, if player relogin join normal.
> For @Si1ker images this is a normal mode no bug. Comerce, lfgroup, lfguild channels only work in city.

yes, that's the problem I'm pointing out, it's at the first connection you don't reach /1 and /3, you have to disconnect and reconnect to be able to have those channels there. I'm not talking about the cities, but about the starting points at the first connection of the character in question.

So @FrancescoBorzi u can remove "Not confirmed" tag :)

**Comment by Si1ker** [comment]
@solidmaxtor Yeah I know how it works, I said, I'm getting all channels while in town/city, just like an extra note.
@Maurowin First-ever login with every single race and I got channels 1 and 3, as you can see in the pictures. Tha's what you said "it's not working" on this issue.

**Comment by Maurowin** [comment]
There are two of us who have this problem, I don't know how to explain it in any case.

I recompiled several b868f0a on a VM especially dedicated to AC

**Comment by Helias** [comment]
If one of us with the same core get a error, and another one with the same core doesn't, probably the error could be related to the client or server modules (which maybe we forgot to remove/clean) but not to the core I think,

**Comment by BarbzYHOOL** [comment]
maxtor you're on linux too?

**Comment by Helias** [comment]
Ubuntu 18.04, I got the same issue.

![image](https://user-images.githubusercontent.com/519778/77565163-1e58c180-6ec4-11ea-8e06-0fd30edcdf60.png)

**Comment by BarbzYHOOL** [comment]
I bet Silker is on windows and Maxtor on linux, so it could possibly be a linux only issue o_0

**Comment by Helias** [comment]
I am thinking about worldserver.conf instead.

**Comment by Viste** [comment]
i think its bad itr in locale choose

**Comment by Helias** [comment]
Yes, it's the client.

I just tested this with a esES client (should be the same of @solidmaxtor ) and I didn't get the issue.

![image](https://user-images.githubusercontent.com/519778/77567588-ad1b0d80-6ec7-11ea-904b-7c4b7ee8a383.png)

So I am closing this issue because is not related to the server.

**Comment by BarbzYHOOL** [comment]
what? i didn't understand

**Comment by Helias** [comment]
the bug depends on the Client, on esES it does not happen, in enUS yes, but it's not an AC bug

**Comment by Maurowin** [comment]
@Helias i use same client to AC and Warmane, and i have no issue with Warmane. i think we need to search a little bit

**Comment by Helias** [comment]
Ok, I will keep this open, we need to investigate and understand if the server can improve this kind of bug

**Comment by Viste** [comment]
> i think its bad itr in locale choose

this

**Comment by Maurowin** [comment]
Up :)

**Comment by Maurowin** [comment]
> 
> 
> > i think its bad itr in locale choose
> 
> this

For me Viste is right

**Comment by Viste** [comment]
can someone show account table with bugged account

**Comment by Maurowin** [comment]
> 
> 
> can someone show account table with bugged account

I don't understand. It doesn't depend on an account but on the language of the client used.

**Comment by Viste** [comment]
> > can someone show account table with bugged account
> 
> I don't understand. It doesn't depend on an account but on the language of the client used. In my case FR and EN

first time when server deteccts locale is in Auth
```
CREATE TABLE `account` 
(
  `id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Identifier',
  `username` varchar(32) NOT NULL DEFAULT '',
  `sha_pass_hash` varchar(40) NOT NULL DEFAULT '',
  `sessionkey` varchar(80) NOT NULL DEFAULT '',
  `v` varchar(64) NOT NULL DEFAULT '',
  `s` varchar(64) NOT NULL DEFAULT '',
  `token_key` varchar(100) NOT NULL DEFAULT '',
  `email` varchar(255) NOT NULL DEFAULT '',
  `reg_mail` varchar(255) NOT NULL DEFAULT '',
  `joindate` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
  `last_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
  `last_attempt_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
  `failed_logins` int(10) unsigned NOT NULL DEFAULT '0',
  `locked` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `lock_country` varchar(2) NOT NULL DEFAULT '00',
  `last_login` timestamp NULL DEFAULT NULL,
  `online` tinyint(3) unsigned NOT NULL DEFAULT '0',
  `expansion` tinyint(3) unsigned NOT NULL DEFAULT '2',
  `mutetime` bigint(20) NOT NULL DEFAULT '0',
  `mutereason` varchar(255) NOT NULL DEFAULT '',
  `muteby` varchar(50) NOT NULL DEFAULT '',
  `**locale**` tinyint(3) unsigned NOT NULL DEFAULT '0',```

**Comment by Maurowin** [comment]
> 
> 
> > > can someone show account table with bugged account
> > 
> > 
> > I don't understand. It doesn't depend on an account but on the language of the client used. In my case FR and EN
> 
> first time when server deteccts locale is in Auth
> 
> ```
> CREATE TABLE `account` 
> (
>   `id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Identifier',
>   `username` varchar(32) NOT NULL DEFAULT '',
>   `sha_pass_hash` varchar(40) NOT NULL DEFAULT '',
>   `sessionkey` varchar(80) NOT NULL DEFAULT '',
>   `v` varchar(64) NOT NULL DEFAULT '',
>   `s` varchar(64) NOT NULL DEFAULT '',
>   `token_key` varchar(100) NOT NULL DEFAULT '',
>   `email` varchar(255) NOT NULL DEFAULT '',
>   `reg_mail` varchar(255) NOT NULL DEFAULT '',
>   `joindate` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
>   `last_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
>   `last_attempt_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
>   `failed_logins` int(10) unsigned NOT NULL DEFAULT '0',
>   `locked` tinyint(3) unsigned NOT NULL DEFAULT '0',
>   `lock_country` varchar(2) NOT NULL DEFAULT '00',
>   `last_login` timestamp NULL DEFAULT NULL,
>   `online` tinyint(3) unsigned NOT NULL DEFAULT '0',
>   `expansion` tinyint(3) unsigned NOT NULL DEFAULT '2',
>   `mutetime` bigint(20) NOT NULL DEFAULT '0',
>   `mutereason` varchar(255) NOT NULL DEFAULT '',
>   `muteby` varchar(50) NOT NULL DEFAULT '',
>   `locale` tinyint(3) unsigned NOT NULL DEFAULT '0', -------------------------- THIS
> ```

ooh ok, do u need structure or datas ?

**Comment by Viste** [comment]
> > > > can someone show account table with bugged account
> > > 
> > > 
> > > I don't understand. It doesn't depend on an account but on the language of the client used. In my case FR and EN
> > 
> > 
> > first time when server deteccts locale is in Auth
> > ```
> > CREATE TABLE `account` 
> > (
> >   `id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Identifier',
> >   `username` varchar(32) NOT NULL DEFAULT '',
> >   `sha_pass_hash` varchar(40) NOT NULL DEFAULT '',
> >   `sessionkey` varchar(80) NOT NULL DEFAULT '',
> >   `v` varchar(64) NOT NULL DEFAULT '',
> >   `s` varchar(64) NOT NULL DEFAULT '',
> >   `token_key` varchar(100) NOT NULL DEFAULT '',
> >   `email` varchar(255) NOT NULL DEFAULT '',
> >   `reg_mail` varchar(255) NOT NULL DEFAULT '',
> >   `joindate` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
> >   `last_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
> >   `last_attempt_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
> >   `failed_logins` int(10) unsigned NOT NULL DEFAULT '0',
> >   `locked` tinyint(3) unsigned NOT NULL DEFAULT '0',
> >   `lock_country` varchar(2) NOT NULL DEFAULT '00',
> >   `last_login` timestamp NULL DEFAULT NULL,
> >   `online` tinyint(3) unsigned NOT NULL DEFAULT '0',
> >   `expansion` tinyint(3) unsigned NOT NULL DEFAULT '2',
> >   `mutetime` bigint(20) NOT NULL DEFAULT '0',
> >   `mutereason` varchar(255) NOT NULL DEFAULT '',
> >   `muteby` varchar(50) NOT NULL DEFAULT '',
> >   `locale` tinyint(3) unsigned NOT NULL DEFAULT '0', -------------------------- THIS
> > ```
> 
> ooh ok, do u need structure or datas ?

i already send you table struct ьaybe I need it, yeah....

**Comment by Maurowin** [comment]
`'11', 'MAUROWIN', 'hash of my password (deleted intentionaly)', 'hash of my session (deleted intentionaly)', 'hash of my "v" column (deleted intentionaly)', 'hash of my "s" column (deleted intentionaly)', '', '', '', '2020-05-11 00:07:16', '192.168.1.33', '192.168.1.33', '0', '0', '00', '2020-11-16 22:59:08', '0', '2', '0', '', '', '0', 'Win', '0', '9120'
`

**Comment by Maurowin** [comment]
> 
> 
> > > > > can someone show account table with bugged account
> > > > 
> > > > 
> > > > I don't understand. It doesn't depend on an account but on the language of the client used. In my case FR and EN
> > > 
> > > 
> > > first time when server deteccts locale is in Auth
> > > ```
> > > CREATE TABLE `account` 
> > > (
> > >   `id` int(10) unsigned NOT NULL AUTO_INCREMENT COMMENT 'Identifier',
> > >   `username` varchar(32) NOT NULL DEFAULT '',
> > >   `sha_pass_hash` varchar(40) NOT NULL DEFAULT '',
> > >   `sessionkey` varchar(80) NOT NULL DEFAULT '',
> > >   `v` varchar(64) NOT NULL DEFAULT '',
> > >   `s` varchar(64) NOT NULL DEFAULT '',
> > >   `token_key` varchar(100) NOT NULL DEFAULT '',
> > >   `email` varchar(255) NOT NULL DEFAULT '',
> > >   `reg_mail` varchar(255) NOT NULL DEFAULT '',
> > >   `joindate` timestamp NOT NULL DEFAULT CURRENT_TIMESTAMP,
> > >   `last_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
> > >   `last_attempt_ip` varchar(15) NOT NULL DEFAULT '127.0.0.1',
> > >   `failed_logins` int(10) unsigned NOT NULL DEFAULT '0',
> > >   `locked` tinyint(3) unsigned NOT NULL DEFAULT '0',
> > >   `lock_country` varchar(2) NOT NULL DEFAULT '00',
> > >   `last_login` timestamp NULL DEFAULT NULL,
> > >   `online` tinyint(3) unsigned NOT NULL DEFAULT '0',
> > >   `expansion` tinyint(3) unsigned NOT NULL DEFAULT '2',
> > >   `mutetime` bigint(20) NOT NULL DEFAULT '0',
> > >   `mutereason` varchar(255) NOT NULL DEFAULT '',
> > >   `muteby` varchar(50) NOT NULL DEFAULT '',
> > >   `locale` tinyint(3) unsigned NOT NULL DEFAULT '0', -------------------------- THIS
> > > ```
> > 
> > 
> > ooh ok, do u need structure or datas ?
> 
> i already send you table struct... I need it, yeah....
> data ofc

With EN client :
`'11', 'MAUROWIN', 'hash of my password (deleted intentionaly)', 'hash of my session (deleted intentionaly)', 'hash of my "v" column (deleted intentionaly)', 'hash of my "s" column (deleted intentionaly)', '', '', '', '2020-05-11 00:07:16', '192.168.1.33', '192.168.1.33', '0', '0', '00', '2020-11-18 17:25:21', '0', '2', '0', '', '', '0', 'Win', '0', '9120'
`

With FR client : 
`'11', 'MAUROWIN', 'hash of my password (deleted intentionaly)', 'hash of my session (deleted intentionaly)', 'hash of my "v" column (deleted intentionaly)', 'hash of my "s" column (deleted intentionaly)', '', '', '', '2020-05-11 00:07:16', '192.168.1.33', '192.168.1.33', '0', '0', '00', '2020-11-18 17:31:39', '0', '2', '0', '', '', '2', 'Win', '0', '9120'
`
